fendermint/vm/interpreter/src/fvm/exec.rs

```diff
@@ -83,6 +83,31 @@ where
             anyhow::bail!("failed to apply block cron message: {}", err);
         }
 
+        // Push the current block hash to the chainmetadata actor
+        //
+        let params = fvm_ipld_encoding::RawBytes::serialize(
+            fendermint_actor_chainmetadata::PushBlockParams {
+                epoch: height,
+                block: fendermint_vm_message::cid(&state.block_hash().unwrap()).unwrap(),
```

Comment:
`fendermint_vm_message::cid` serializes its input and hashes it to produce a CID. That's a one-way conversion so if we wanted to use what we stored to actually identify blocks, say, to retrieve them from the API, we'd be in trouble because while this value would be unique, it wouldn't be recognised by the CometBFT API. 

I think we could _pretend_ that the `[u8; 32]` returned by `block_hash()` is a CID. It is after all a SHA256 (I just don't know if't it's Sha2 or Sha3; I have a `tx_hash` method but it's unclear) of the block, as given to us by CometBFT, and it's certainly possible for a CID to store this.

Then we can unwrap it to get to the raw digest if we have to use it with the CometBFT API.

Reply:
Updated to store only the blockhash digest in the state